src/components/Elements.js

```diff
@@ -102,44 +119,38 @@ export const Elements = ({stripe, options, children}: Props) => {
 
 Elements.propTypes = {
   stripe: PropTypes.any,
-  children: PropTypes.any,
+  children: PropTypes.node,
   options: PropTypes.object,
 };
 
-const parseElementsContext = (
-  ctx: ?ElementContext,
-  useCase: string
-): null | ElementsShape => {
-  if (!ctx) {
-    throw new Error(
-      `Could not find elements context; You need to wrap the part of your app that is ${useCase} in an <Elements> provider.`
-    );
-  }
+export const useElementsContextWithUseCase = (useCaseMessage: string) => {
+  const ctx = useContext(ElementsContext);
+  return parseElementsContext(ctx, useCaseMessage);
+};
 
-  if (ctx.tag === 'loading') {
-    return null;
-  }
+export const useElements = (): ElementsShape | null => {
+  const ctx = useElementsContextWithUseCase('calls useElements()');
 
-  return ctx.elements;
+  return ctx && ctx.elements;
 };
 
-export const useElementsWithUseCase = (useCaseMessage: string) => {
-  const ctx = useContext(ElementsContext);
-  return parseElementsContext(ctx, useCaseMessage);
-};
+export const useStripe = (): StripeShape | null => {
+  const ctx = useElementsContextWithUseCase('calls useStripe()');
 
-export const useElements = () =>
-  useElementsWithUseCase('calling useElements()');
+  return ctx && ctx.stripe;
```

Comment:
I wonder if `<Elements>` should avoid rendering its children until a valid `stripe` prop is passed. It would be nice to not check for `null` when using `useStripe`.

Reply:
(as a user of the library)

Reply:
Oh man that would be a big change, but at that point we ought to just make Stripe a required prop and tell people not to render Elements until they have it.

Reply:
If a user wraps the whole checkout form in Elements, they may want it to render while Stripe is loading and just use a placeholder for the card field. Not having that option is a pretty big drawback.